### ISSUE TITLE
Function selectors from inherited interfaces are not visible in parent interface

### ISSUE THREAD

**Issue by mnusurov** [ethereum/solidity#15157]
## Description

The following code fails to compile with: 

> TypeError: Member "b" not found or not visible after argument-dependent lookup in type(contract I2).

```
pragma solidity 0.8.26;

interface I1 {
    function b() external;
}

interface I2 is I1 {
    function a() external;
}

// ✅ Test 1 -- passes 
contract test_a {
    function a() external view returns (bytes4) {
        return I2.a.selector;
    }
}

// ❌ Test 1 -- should pass
contract test_b {
    function b() external view returns (bytes4) {
        return I2.b.selector;
    }
}
```

## Why?

According to [official v0.8.26 documentation](https://docs.soliditylang.org/en/v0.8.26/contracts.html#interfaces), 

> Interfaces can inherit from other interfaces. This has the same rules as normal inheritance:

**Comment by nikola-matic** [comment]
Thanks for the report @mnusurov.

**Comment by smartcontracts** [comment]
Yeah just ran into this too, would be a good quality of life improvement

**Comment by bunbun295** [comment]
we ran into this for @0xTwyne contracts. Would be good to fix this.

**Comment by cameel** [comment]
> According to [official v0.8.26 documentation](https://docs.soliditylang.org/en/v0.8.26/contracts.html#interfaces), 
>
> > Interfaces can inherit from other interfaces. This has the same rules as normal inheritance:

We don't have a a documented rule that says that definitions must be available via derived contract name. In fact the sentence just below this part says:

> Types defined inside interfaces and other contract-like structures can be accessed from other contracts: `Token.TokenType` or `Token.Coin`.

with `Token` being the interface, not a contract derived from it.

In that sense I would not say it's a bug. The language does not allow accessing events, errors (#13656), enums, structs, UDVTs (#15902) or constants this way either. See the example in https://github.com/argotorg/solidity/issues/15902#issuecomment-5022296619.

Having said that, this is still a valid feature request. I think it would make sense to change this behavior both for functions and for all other cases I mentioned. This should be discussed on one of the design calls.

**Comment by cameel** [comment]
We have another issue that covers this as well (#16840). It was submitted later, but is a bit broader and more complete (properly considers the backwards-compatibility issues) so I'm closing this one as a duplicate.
